rust/chains/tw_ton/src/signing_request/builder.rs

```diff
@@ -128,11 +128,20 @@ impl SigningRequestBuilder {
             .into_tw()
             .context("Invalid 'response_address' address")?;
 
+        let custom_payload = input
+            .custom_payload
+            .as_ref()
+            .map(|payload| TransferCustomRequest {
+                state_init: Some(payload.state_init.to_string()),
+                payload: Some(payload.payload.to_string()),
+            });
```

Comment:
Either `payload.state_init` or `payload.payload` can be empty. In that case, we consider them as None.
There is `tw_misc::traits::OptionalEmpty` that you can use for that purpose.